src/components/data-table/ha-data-table.ts

```diff
@@ -233,6 +237,14 @@ export class HaDataTable extends LitElement {
       // Force update of location of rows
       this._filteredData = [...this._filteredData];
     }
+
+    for (const columnId in this.columns) {
+      if (this.columns[columnId].direction) {
+        this._defaultSortDirection = this.columns[columnId].direction!;
+        this._defaultSortColumn = columnId;
+        break;
+      }
+    }
```

Comment:
```suggestion
  if (changedProperties.has("columns")) {
    for (const columnId in this.columns) {
      if (this.columns[columnId].direction) {
        this._defaultSortDirection = this.columns[columnId].direction!;
        this._defaultSortColumn = columnId;
        break;
      }
    }
  }
```
So we don't check this for every render

Reply:
But this initialisation occurs only in connectedCallback(), not in firstUpdated() or updated().
Should move it to firstUpdated() ?

Reply:
Removed my comment by mistake, restoring:
These operations are added into `connectedCallback`, shall I move them into `firstUpdated()` ?

Reply:
Also, if I move this to `firstUpdated` - then probably I do not need to check for `changedProperties.has("columns")`, right?
Or I should need to move it to `updated()` instead? (not sure...)

Reply:
Oh I though this was in `willUpdate`. Sorry. This is fine then